File: Software/Bit-button/python/bit-button.py
#!/usr/bin/env python3
# -*- coding: iso-8859-15 -*-
from serial import Serial

SERIAL_PORT = "/dev/ttyUSB1"

class bit_button:
    def __init__(self):
        self.sp = Serial(SERIAL_PORT, 115200)

    def bit(self, b):
        """Bit to Send
          0: send the '0\n' string
          1: send the '1\n' string
        """
        if b in [0,1]:
          self.sp.write(str(b).encode())
          self.sp.write(b'\n')


# -- Example
input_pin = bit_button()

# -- Send a 1
input_pin.bit(1)

# -- Ask the user for a bit
while True:
    b = int(input("Enter a bit value (0/1): "))
    input_pin.bit(b)

File: Software/Bit-button/nodejs/bit-button.js
const SerialPort = require('serialport');
const port = new SerialPort('/dev/ttyUSB1', {
  baudRate:115200
});

console.log("Enter the bit value + ENTER");

// -- Send 1
port.write('1\n');

//-- When the user input something...
process.stdin.on('data', (b) => {

  //-- Convert into a string
  let sb = String(b);

  //-- Only strings "1\n" and "0\n" are valid
  if (sb == "1\n" || sb == "0\n") {
    console.log("Bit: " + sb[0]);

    //-- Transmit the bit
    port.write(b);
  }

});
